controllers/openshift_controller.go

```diff
@@ -1440,6 +1482,31 @@ func (r *KataConfigOpenShiftReconciler) labelNodes(nodeSelector *metav1.LabelSel
 
 }
 
+func (r *KataConfigOpenShiftReconciler) unlabelNodes(nodeSelector *metav1.LabelSelector) (err error) {
+	labelSelector, _ := metav1.LabelSelectorAsSelector(nodeSelector)
+	nodeList := &corev1.NodeList{}
+	listOpts := []client.ListOption{
+		client.MatchingLabelsSelector{Selector: labelSelector},
+	}
+
+	if err := r.Client.List(context.TODO(), nodeList, listOpts...); err != nil {
+		r.Log.Error(err, "Getting list of nodes failed")
+		return err
+	}
+
+	for _, node := range nodeList.Items {
+		if _, ok := node.Labels["node-role.kubernetes.io/kata-oc"]; ok {
+			delete(node.Labels, "node-role.kubernetes.io/kata-oc")
+			err = r.Client.Update(context.TODO(), &node)
+			if err != nil {
+				r.Log.Error(err, "Error when removing labels from node", "node", node)
+				return err
+			}
```

Comment:
I considered that but it doesn't really seem all that beneficial.  `unlabelNode()` has two lines of substantial code, the rest is error handling. But since it can still return an error, the error handling here, at the call site, would need to remain.  So using `unlabelNode()` would save a single line here but would require us to keep the whole `unlabelNode()` function.

Additionally, given that `unlabelNode()` will have no other callers, I really think that doing the unlabeling here and removing `unlabelNode()` in the final commit of this PR is a better solution.

Reply:
I agree that `unlabelNode()` doesn't bring much and could have been open-coded from the start. I guess I would just say something along those lines in the commit message : useless helper, better done open-coded, will be removed later.

Reply:
I think it _is_ mentioned in the commit message, isn't it? In the final paragraph I mean.

Reply:
Well not exactly but nevermind. Let's go forward ! :wink: